### PR TITLE
Bump xo@0.59.3

### PR DESCRIPTION
**Pull request by LitoMore**
Upgrade XO to the latest version. It resolves the deprecation warning:

```
(node:17770) [DEP0180] DeprecationWarning: fs.Stats constructor is deprecated.
(Use `node --trace-deprecation ...` to show where the warning was created)
```

### Other errors

I also fixed those new errors from the latest XO.

```
❯ npx xo
  sdk.mjs:6:1
  ✖    6:1   The fs/promises is still an experimental feature and is not supported until Node.js 14.13.1. The configured version range is >=14.                                  n/no-unsupported-features/node-builtins
  ✖    7:1   The path is still an experimental feature and is not supported until Node.js 14.13.1 (backported: ^12.20.0). The configured version range is >=14.                  n/no-unsupported-features/node-builtins
  ✖    8:1   The url is still an experimental feature and is not supported until Node.js 14.13.1 (backported: ^12.20.0). The configured version range is >=14.                   n/no-unsupported-features/node-builtins

  scripts/build/clean.js:18:20
  ✖   18:20  The fs/promises.constants is still an experimental feature and is not supported until Node.js 18.4.0 (backported: ^16.17.0). The configured version range is >=18.  n/no-unsupported-features/node-builtins
  ✖   19:6   Prefer await to then()/catch()/finally().                                                                                                                           promise/prefer-await-to-then
  ✖   20:6   Prefer await to then()/catch()/finally().                                                                                                                           promise/prefer-await-to-then

  sdk.d.ts:6:47
  ✖    6:47  require file extension .js.                                                                                                                                         n/file-extension-in-import

  scripts/build/package.js:14:1
  ✖   14:1   Use named import for module node:util.                                                                                                                              unicorn/import-style

  8 errors
```

### CODE REVIEW

File: scripts/build/clean.js [hunk 1]
```diff
@@ -13,11 +13,14 @@ const __dirname = getDirnameFromImportMeta(import.meta.url);
...
 const files = ['index.js', 'index.mjs', 'index.d.ts', 'sdk.js'];
 
-const fileExists = (fpath) =>
-  fs
-    .access(fpath, fs.constants.F_OK)
-    .then(() => true)
-    .catch(() => false);
+const fileExists = async (fpath) => {
+  try {
+    await fs.access(fpath, fs.constants.F_OK);
```
Comment (mondeja) [inline scripts/build/clean.js:18]: We can just remove `fs.constants.F_OK` as [is the default value](https://nodejs.org/api/fs.html#fspromisesaccesspath-mode) and a bit verbose I guess.

```suggestion
    await fs.access(fpath);
```

File: .xo-config.json [hunk 2]
```diff
@@ -47,7 +47,7 @@
         "svglint.config.mjs",
         "svgo.config.mjs"
       ],
-      "nodeVersion": ">=18"
+      "nodeVersion": ">=18.4"
```
Comment (mondeja) [inline .xo-config.json:50]: So `nodeVersion` could still be pinned at `>=18`.

Comment (LitoMore) [reply .xo-config.json:50]: > The fs/promises.constants is still an experimental feature and is not supported until Node.js 18.4.0.

Comment (LitoMore) [reply .xo-config.json:50]: Ah, I see.

**Review by mondeja — APPROVED** [review]
Great, thanks! 🥇

**Pull request merged**